src/lib/tls/tls_algos.h

```diff
@@ -199,8 +203,13 @@ class BOTAN_PUBLIC_API(3, 2) Group_Params final {
       }
 
       constexpr bool is_pure_kyber() const {
+         BOTAN_DIAGNOSTIC_PUSH
+         BOTAN_DIAGNOSTIC_IGNORE_DEPRECATED_DECLARATIONS
+
          return m_code == Group_Params_Code::KYBER_512_R3_OQS || m_code == Group_Params_Code::KYBER_768_R3_OQS ||
                 m_code == Group_Params_Code::KYBER_1024_R3_OQS;
+
+         BOTAN_DIAGNOSTIC_POP
       }
```

Comment:
Probably reasonable deprecating this as well. Also regarding https://github.com/randombit/botan/pull/4305#discussion_r1814643378.
Note that `is_kem()` would then probably also need the `BOTAN_DIAGNOSTIC_IGNORE_DEPRECATED_DECLARATIONS` treatment 😞.

```suggestion
      BOTAN_TLS_KYBER_R3_DEPRECATED constexpr bool is_pure_kyber() const {
         BOTAN_DIAGNOSTIC_PUSH
         BOTAN_DIAGNOSTIC_IGNORE_DEPRECATED_DECLARATIONS

         return m_code == Group_Params_Code::KYBER_512_R3_OQS || m_code == Group_Params_Code::KYBER_768_R3_OQS ||
                m_code == Group_Params_Code::KYBER_1024_R3_OQS;

         BOTAN_DIAGNOSTIC_POP
      }
```